File: Coursework.API/Core/Entities/Origin/Game.cs
using Core.Models;
using System;

namespace Core.Entities.Origin
{
    class Game : IEntity
    {
        public string Id { get; set; }

        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime DateReleased { get; set; }
    }
}

File: coursework-front-end/src/app/models/game.ts
import { IModel } from './imodel';

export class Game implements IModel {
    constructor(
        public id?: string,
        public name?: string,
        public description?: string,
        public dateReleased?: Date,
    ) {}

    public static mock(): Game {
        return new Game("Mock", "Mock", "That's placeholder", new Date());
    }
}